tooling/debugger/src/dap.rs

```diff
@@ -213,37 +231,40 @@ impl<'a, R: Read, W: Write, B: BlackBoxFunctionSolver> DapSession<'a, R, W, B> {
         Ok(())
     }
 
+    fn build_stack_trace(&self) -> Vec<StackFrame> {
+        self.context
+            .get_source_call_stack()
+            .iter()
+            .enumerate()
+            .map(|(index, (opcode_location, source_location))| {
+                let line_number =
+                    self.debug_artifact.location_line_number(*source_location).unwrap();
+                let column_number =
+                    self.debug_artifact.location_column_number(*source_location).unwrap();
+                StackFrame {
+                    id: index as i64,
+                    name: format!("frame #{index}"),
+                    source: Some(Source {
+                        path: self.debug_artifact.file_map[&source_location.file]
+                            .path
+                            .to_str()
+                            .map(String::from),
+                        ..Source::default()
+                    }),
+                    line: line_number as i64,
+                    column: column_number as i64,
+                    instruction_pointer_reference: Some(opcode_location.to_string()),
+                    ..StackFrame::default()
+                }
+            })
+            .collect::<Vec<_>>()
+            .into_iter()
+            .rev()
```

Comment:
I think you could probably avoid this `rev` by having a `Vec<StackFrame` outside the map and appropriately setting it while mapping the source call stack. This is a nit though and the `rev` is more functional.

Reply:
The first `.collect()` wasn't necessary, so I removed it. The `.rev()` is still there, but it now reverses the `Map` iterator, so it reads better.